src/adaXT/decision_tree/decision_tree.pyx

```diff
@@ -155,24 +155,28 @@ class DecisionTree(BaseModel):
         return ht
 
     def _tree_based_weights(self, hash0: dict, hash1: dict, size_X0: int,
-                            size_X1: int, scale: bool) -> np.ndarray:
+                            size_X1: int, scaling: str) -> np.ndarray:
         matrix = np.zeros((size_X0, size_X1))
         hash0_keys = hash0.keys()
         hash1_keys = hash1.keys()
         for xi in hash0_keys:
             if xi in hash1_keys:
                 indices_1 = hash0[xi]
                 indices_2 = hash1[xi]
-                if scale == "column":
-                    val = 1.0/len(indices_1)
-                elif scale == "symmetric":
-                    val = 1.0/(len(indices_1) + len(indices_2))
-                elif scale == "none":
+                if scaling == "row":
+                    val = 1.0/len(indices_2)
+                    for ind2 in indices_2:
+                        matrix[indices_1, ind2] += val
+                elif scaling == "similarity":
                     val = 1.0
-                matrix[np.ix_(indices_1, indices_2)] = val
+                    matrix[np.ix_(indices_1, indices_2)] = val
+                elif scaling == "none":
+                    val = 1.0
+                    for ind2 in indices_2:
+                        matrix[indices_1, ind2] += val
         return matrix
 
-    def similarity(self, X0: ArrayLike, X1: ArrayLike, scale: bool = True):
+    def similarity(self, X0: ArrayLike, X1: ArrayLike):
```

Comment:
We don't want there to be no scaling for similarity any more?

Reply:
No, I couldnt come up with an example why one would want to do this so I thought we might as well just remove it.